moveit_ros/moveit_servo/src/utils/command.cpp

```diff
@@ -272,6 +273,24 @@ JointDeltaResult jointDeltaFromIK(const Eigen::VectorXd& cartesian_position_delt
     delta_theta = pseudo_inverse * cartesian_position_delta;
   }
 
+  if (!servo_params.active_subgroup.empty())
+  {
+    // Create full delta vector and add delta theta's at actuated joints
+    const auto& move_group_joint_names =
+        robot_state->getJointModelGroup(servo_params.move_group_name)->getActiveJointModelNames();
+    const auto& subgroup_joint_names =
+        robot_state->getJointModelGroup(servo_params.active_subgroup)->getActiveJointModelNames();
+
+    Eigen::VectorXd move_group_delta_theta = Eigen::VectorXd::Zero(move_group_joint_names.size());
+    for (size_t index = 0; index < subgroup_joint_names.size(); index++)
+    {
+      const auto move_group_iterator =
+          find(move_group_joint_names.cbegin(), move_group_joint_names.cend(), subgroup_joint_names.at(index));
+      move_group_delta_theta[std::distance(move_group_joint_names.cbegin(), move_group_iterator)] = delta_theta[index];
```

Comment:
I can see this adding a lot of slowness to the regular running loop. I wonder if we should instead build an unordered map either at initialization (go through all the subgroups?) and/or at parameter switching time so it could let us look up indices in O(1) time in the loop?

... or if there is otherwise already another way of doing this with `RobotModel`?

Reply:
> at parameter switching time 

This option looks better since we are in a free function.
If we create map at initialization, it would likely need to be passed through multiple functions.

Reply:
I moved this find operation into the param update function so it is only called once and only the first time a group is added. We could even think about moving it into the constructor :thinking: and just create a map for all existing joint groups

Reply:
Done :heavy_check_mark: